### ISSUE TITLE
new testcase is failing after 3f134877 on mec1501modular_assy6885

### ISSUE THREAD

**Issue by yingmingx** [zephyrproject-rtos/zephyr#30809]
The issue:

miss macro TIMEOUT after commit-id 3f134877 
new testcase(tests/drivers/sensor/generic #30763) is failing after 3f134877 on mec1501modular_assy6885:

tests/drivers/sensor/generic/driver.sensor in mec1501modular_assy6885:tests/drivers/sensor/generic/driver.sensor
Including boilerplate (Zephyr base): /workdir/zephyr/cmake/app/boilerplate.cmake
-- Application: /workdir/zephyr/tests/drivers/sensor/generic
-- Zephyr version: 2.4.99 (/workdir/zephyr)
-- Found Python3: /usr/bin/python3.6 (found suitable exact version "3.6.9") found components:  Interpreter 
-- Found west (found suitable version "0.8.0", minimum required is "0.7.1")
-- Board: mec1501modular_assy6885
-- Cache files will be written to: /root/.cache/zephyr
-- Found toolchain: zephyr (/opt/sdk/zephyr-sdk-0.11.4)
-- Found dtc: /opt/sdk/zephyr-sdk-0.11.4/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /workdir/zephyr/boards/arm/mec1501modular_assy6885/mec1501modular_assy6885.dts
-- Generated zephyr.dts: /workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor/zephyr/include/generated/devicetree_unfixed.h
Parsing /workdir/zephyr/Kconfig
Loaded configuration '/workdir/zephyr/boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig'
Merged configuration 'prj.conf'
Configuration saved to '/workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor/zephyr/.config'
Kconfig header saved to '/workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/sdk/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at ../../../../boards/arm/mec1501modular_assy6885/CMakeLists.txt:29 (message):
  Microchip SPI Image Generation tool (everglades_spi_gen_lin64) is not
  available.  SPI Image will not be generated.

-- Configuring done
-- Generating done
-- Build files have been written to: /workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor
[1/143] Preparing syscall dependency handling

[2/143] Generating misc/generated/syscalls_subdirs.trigger
[3/143] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[4/143] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[5/143] Generating include/generated/driver-validation.h
[6/143] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h
[7/143] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[8/143] Generating include/generated/offsets.h
[9/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[10/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[11/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[12/143] Building C object CMakeFiles/app.dir/src/dummy_sensor.c.obj
[13/143] Building C object CMakeFiles/app.dir/src/main.c.obj
[14/143] Linking C static library app/libapp.a
[15/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[16/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[17/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[18/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj
[19/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[20/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[21/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[22/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[23/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[24/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
[25/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[26/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[27/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/work_q.c.obj
[28/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[29/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[30/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[31/143] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[32/143] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[33/143] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/microchip_mec/mec1501/power.c.obj
[34/143] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/microchip_mec/mec1501/soc.c.obj
[35/143] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_minimal.c.obj
[36/143] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/microchip_mec/mec1501/device_power.c.obj
[37/143] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/power/power.c.obj
[38/143] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/console/uart_console.c.obj
[39/143] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/power/policy/policy_residency.c.obj
[40/143] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/timer/sys_clock_init.c.obj
[41/143] Generating linker.cmd
[42/143] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/pinmux/pinmux_mchp_xec.c.obj
[43/143] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj
[44/143] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/timer/mchp_xec_rtos_timer.c.obj
[45/143] Linking C static library zephyr/libzephyr.a
[46/143] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[47/143] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[48/143] Linking C static library zephyr/arch/common/libisr_tables.a
[49/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[50/143] Linking C static library zephyr/arch/common/libarch__common.a
[51/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj
[52/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj
[53/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj
[54/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj
[55/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj
[56/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj
[57/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj
[58/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[59/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/vector_table.S.obj
[60/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault_s.S.obj
[61/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/reset.S.obj
[62/143] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/exc_exit.S.obj
[63/143] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj
[64/143] Linking C static library zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a
[65/143] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
[66/143] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/irq_init.c.obj
[67/143] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault.c.obj
[68/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/abort.c.obj
[69/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/atoi.c.obj
[70/143] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/thread_abort.c.obj
[71/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtol.c.obj
[72/143] Linking C static library zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a
[73/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoul.c.obj
[74/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/bsearch.c.obj
[75/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strncasecmp.c.obj
[76/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj
[77/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strstr.c.obj
[78/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/exit.c.obj
[79/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/stdout_console.c.obj
[80/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strspn.c.obj
[81/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/sprintf.c.obj
[82/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/fprintf.c.obj
[83/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/string.c.obj
[84/143] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/time/gmtime.c.obj
[85/143] Linking C static library zephyr/lib/libc/minimal/liblib__libc__minimal.a
[86/143] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
[87/143] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/nanosleep.c.obj
[88/143] Linking C static library zephyr/lib/posix/liblib__posix.a
[89/143] Building C object zephyr/boards/arm/mec1501modular_assy6885/CMakeFiles/boards__arm__mec1501modular_assy6885.dir/pinmux.c.obj
[90/143] Building C object zephyr/drivers/adc/CMakeFiles/drivers__adc.dir/adc_common.c.obj
[91/143] Linking C static library zephyr/boards/arm/mec1501modular_assy6885/libboards__arm__mec1501modular_assy6885.a
[92/143] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj
[93/143] Linking C static library zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a
[94/143] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_common.c.obj
[95/143] Building C object zephyr/drivers/adc/CMakeFiles/drivers__adc.dir/adc_mchp_xec.c.obj
[96/143] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mchp_xec.c.obj
[97/143] Linking C static library zephyr/drivers/adc/libdrivers__adc.a
[98/143] Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a
[99/143] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mchp_xec.c.obj
FAILED: zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mchp_xec.c.obj 
ccache /opt/sdk/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.4.0-2501-gdeefb0ac1e8f -DKERNEL -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/arm/microchip_mec/mec1501 -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/cmsis/CMSIS/Core/Include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/microchip/mec/common -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/microchip/mec/mec1501 -isystem ../../../../../../../lib/libc/minimal/include -isystem /opt/sdk/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include -isystem /opt/sdk/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include-fixed -Os -imacros /workdir/zephyr/twister-out/mec1501modular_assy6885/tests/drivers/sensor/generic/driver.sensor/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros /workdir/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/workdir/zephyr/tests/drivers/sensor/generic=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workdir/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workdir=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mchp_xec.c.obj -MF zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mchp_xec.c.obj.d -o zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mchp_xec.c.obj   -c /workdir/zephyr/drivers/i2c/i2c_mchp_xec.c
/workdir/zephyr/drivers/i2c/i2c_mchp_xec.c: In function 'xec_spin_yield':
/workdir/zephyr/drivers/i2c/i2c_mchp_xec.c:127:17: error: 'TIMEOUT' undeclared (first use in this function); did you mean 'ETIMEDOUT'?
  127 |  if (*counter > TIMEOUT) {
      |                 ^~~~~~~
      |                 ETIMEDOUT
/workdir/zephyr/drivers/i2c/i2c_mchp_xec.c:127:17: note: each undeclared identifier is reported only once for each function it appears in
[100/143] Building C object zephyr/drivers/sensor/mchp_tach_xec/CMakeFiles/drivers__sensor__mchp_tach_xec.dir/tach_mchp_xec.c.obj
[101/143] Building C object zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mchp_xec.c.obj
ninja: build stopped: subcommand failed.

**Comment by cfriedt** [comment]
duplicate of #30805
